### DISCUSSION TITLE
Caching

### DISCUSSION THREAD

**Discussion by sohank1** [Automattic/mongoose#11404] [Q&A] [answered]
Does mongoose cache data so that it returns data from memory and only fetches it if needed?

**Comment by Uzlopak** [comment]
This would be kind of critical, dont you think? 

If it would cache data, then you could never trust mongoose data, as it would be potentially always supply you with old data.

**Reply by sohank1** [reply]
Couldn't u use mongo db trigger function so it's always up to date?

**Comment by vkarpov15** [comment, marked as answer]
Nope, Mongoose doesn't cache by default. You're welcome to implement caching, but we tend to find it unnecessary given the way we recommend structuring data.
